pip/src/interpreter.rs

```diff
@@ -828,3 +991,67 @@ where
         PyException::new_err(message)
     }
 }
+
+#[pyclass]
+#[derive(Clone, Copy)]
+struct GlobalCallable {
+    id: StoreItemId,
+}
+
+/// Create a Python callable from a Q# callable and adds it to the given environment.
+fn create_py_callable(
+    py: Python,
+    env: &PyObject,
+    make_callable: &PyObject,
+    namespace: &[Rc<str>],
+    name: &str,
+    item_id: StoreItemId,
+) -> PyResult<()> {
+    if namespace.is_empty() && name == "<lambda>" {
+        // We don't want to bind auto-generated lambda callables.
+        return Ok(());
+    }
+
+    // env is expected to be a module, any other type will raise an error.
+    let mut module: Bound<PyModule> = env.extract(py)?;
```

Comment:
Why are we not passing the env as a pymodule from the beginning? Feels odd that it is a pyobject in the pyi and calling methods.

Reply:
Doing it that way failed compilation on the Rust side, but it turns out @minestarks has a better approach that she is going to push up. With that the module no longer needs to be passed at all, and a bunch of the module manipulation code in Rust goes away.